### DISCUSSION TITLE
why the texture has no color?

### DISCUSSION THREAD

**Discussion by amin2312** [jnsmalm/pixi3d#153] [Q&A]
Hi, sir

I try to build a sphere:

let dirLight = new PIXI3D.Light();
dirLight.type = "directional";
PIXI3D.LightingEnvironment.main.lights.push(dirLight);

const normalTexture = new PIXI3D.StandardMaterialTexture(PIXI.Texture.from('123.png').baseTexture);
const material = new PIXI3D.StandardMaterial();
material.normalTexture = normalTexture;

let model = PIXI3D.Mesh3D.createSphere(material);

the sphere can show texture, but texture don't include any color.
Am I missing some code?

**Comment by jnsmalm** [comment]
Hi,

The normalTexture is only used for lighting. You need to set baseColorTexture for it to show the color like this:

```
let sphere = app.stage.addChild(PIXI3D.Mesh3D.createSphere())
sphere.material.baseColorTexture = PIXI.Texture.from("assets/texture.jpg")
```

**Reply by amin2312** [reply]
thanks for your replay.
but it also don't include any color.
------
let dirLight = new PIXI3D.Light();
dirLight.type = "directional";
PIXI3D.LightingEnvironment.main.lights.push(dirLight);

sphere = StageIns.addChild(PIXI3D.Mesh3D.createSphere());
sphere.material.baseColorTexture = PIXI.Texture.from("i_3d/123.jpg");
sphere.material.normalTexture = PIXI.Texture.from("i_3d/123.jpg");

**Reply by jnsmalm** [reply]
Are you sure those textures are loaded correctly? Correct url?

**Reply by amin2312** [reply]
[test1.zip](https://github.com/jnsmalm/pixi3d/files/10366113/test1.zip)
I make a example, in the zip file, please check it. thank you very much.

**Reply by amin2312** [reply]
![ss](https://user-images.githubusercontent.com/4578021/211148297-57d30294-e9ce-4f3e-9829-993c5a9ec063.jpg)
Its run result in my chrome browse

**Reply by jnsmalm** [reply]
Try removing normalTexture from material as it doesn't contain actual normal data, it will mess things up.

**Reply by amin2312** [reply]
It seems to be a problem with the camera.
I use "CameraOrbitControl" to rotate the stage, then it show the front face.

I just only know write the three.js code:

let camera = new THREE.OrthographicCamera(0, w, 0, h, 1, 1000);
camera.position.set(0, 0, 500);
camera.up.set(0, 0, 0);
camera.lookAt(0, 0, 0);

How to convert it to Pixi3D?

**Reply by jnsmalm** [reply]
Do you mean the texture was black because of camera?

```
PIXI3D.Camera.main.orthographic = true
PIXI3D.Camera.main.position.set(0, 0, 500)
```

**Reply by amin2312** [reply]
It also show the back face. 

![ss](https://user-images.githubusercontent.com/4578021/211152211-6c058689-c623-4db8-afcc-950164071564.jpg)

I use CameraOrbitControl to change the angle, can show it.
But how to make it show the positive face by default

**Reply by jnsmalm** [reply]
You can change the degrees of the angles when using orbit control.

```
let control = new PIXI3D.CameraOrbitControl(app.view)
control.angles.y = 45
```

**Reply by amin2312** [reply]
let control = new PIXI3D.CameraOrbitControl(Renderer2d.view);
control.angles.y = 0

let light = new PIXI3D.Light()
light.type = "directional";
PIXI3D.LightingEnvironment.main.lights.push(light)

PIXI3D.Camera.main.orthographic = true

let sphere = StageIns.addChild(PIXI3D.Mesh3D.createSphere());
sphere.material.baseColorTexture = PIXI.Texture.from("texture_123.jpg");

![ss](https://user-images.githubusercontent.com/4578021/211153040-4b290bc5-cc34-4af0-a2d5-7c280032dbc4.jpg)

It looks normal, but the texture is flipped vertically, how to fix it?

**Reply by jnsmalm** [reply]
The easiest is to just flip the jpg image.

**Reply by amin2312** [reply]
But my texture is normal. Can I flip the camera vertically?

**Comment by jnsmalm** [comment]
The UV coords are set up that way for the built in sphere mesh, you can flip the UV's manually:

```
let sphere = app.stage.addChild(PIXI3D.Mesh3D.createSphere())
for (let i = 1; i < sphere.geometry.uvs[0].buffer.length; i += 2) {
  sphere.geometry.uvs[0].buffer[i] = 1 - sphere.geometry.uvs[0].buffer[i]
}
```

**Reply by amin2312** [reply]
thanks for your help
